### ISSUE TITLE
how to short circuit lintTree

### ISSUE THREAD

**Issue by kellyselden** [ember-cli/ember-cli#4340]
I have this problem where you have a linting addon, but it wants to have a way to still be installed but conditionally disabled. Currently I have something like:
```js
lintTree: function(type, tree) {
  if (myOptions.disabled) {
    return tree;
  }
  return myLintedTree;
}
```
But the short circuit results in code that is not transpiled in my `assets/my-app.js`. Example [here](https://github.com/gdub22/jscs-babel-fail) and Issue [here](https://github.com/kellyselden/broccoli-jscs/issues/31).
I have come up with an ugly solution something like this:
```js
included: function(app) {
  if (!myOptions.disabled) {
    this.lintTree = function(type, tree) {
      return myLintedTree;
    }
  }
}
```
It works, but I wonder if there isn't a better way to handle this, maybe by modifying [addonLintTree](https://github.com/ember-cli/ember-cli/blob/ed8f41ddaef9e1f456b000d1a1b5a283ff5b27c1/lib/broccoli/ember-app.js#L454) from:
```js
if (addon.lintTree) {
  output.push(addon.lintTree(type, tree));
}
```
to something like:
```js
if (addon.lintTree) {
  var newTree = addon.lintTree(type, tree);
  if (newTree !== tree) {
    output.push(newTree);
  }
}
```
or
```js
if (addon.lintTree) {
  var newTree = addon.lintTree(type, tree);
  if (newTree) {
    output.push(newTree);
  }
}
```
If there is a better way, I'm all ears. If this is a good solution, I will submit a PR. I will also update the docs with the answer.

**Comment by kellyselden** [comment]
@ef4 You added this functionality in #3094.
https://github.com/ember-cli/ember-cli/pull/3094/files#diff-392bb207fea8bcbeb4a084bb445f7cc7R340
Do you have any opinion on this?

**Comment by rwjblue** [comment]
I think we should allow lintTree to return falsey, and use a output.filter(Boolean) before calling merge.

**Comment by kellyselden** [comment]
Fixed in #4685
